scripts/export_sqlite_to_workspace.py

```diff
@@ -0,0 +1,259 @@
+    # note: introducting _active flags
+    sett = {}
+    for k, dv in defSett.items():
+
+        parts = k.split("_")
+        cv = csett[k] # current value
+        #print(k, v, cv, parts)
+
+        if parts[1] == "port":
+            if csett.get(k) == "disabled":
+                sett[parts[0] + "_enabled"] = False
+                sett[k] = dv # copy default port no
+            else:
+                sett[parts[0] + "_enabled"] = True
+                sett[k] = int(csett[k])
+        elif k == "pstorage_polling":
+            if csett.get(k) == "disabled":
+                sett[parts[0] + "_enabled"] = False
+                sett[k] = dv  # copy default port no
+            else:
+                sett["pstorage_enabled"] = True
+                sett[k] = int(cv)
+        elif k == "start_run_mode":
+            sett["start_run_mode"] = bool(cv)
+        elif parts[1] == "enabled":
+            sett[k] = bool(cv)
+        else:
+            sett[k] = int(cv)
+
+    write_json(None, "settings.json", sett)
+    write_json(None, "settings.default.json", defSett)
+
+    etc_dir = os.path.abspath(os.path.join(openplc, "etc"))
+    for fn in "active_program", "openplc_driver","openplc_platform":
+        fp = os.path.join(etc_dir, fn)
+        contents = ""
+        with open(fp, "r") as f:
+            contents = f.read()
+            f.close()
+        sf = os.path.join(workspace, "%s.txt" % fn)
+        with open(sf, "w") as f:
+            f.write(contents.strip())
+            f.close()
+            print("Wrote: %s" % sf)
+
+    ## uploads demo
+    up = os.path.join(workspace, "uploads")
+    if not os.path.exists(up):
+        os.mkdir(up)
+
+    st = os.path.join(openplc, "etc", "st_files", "blank_program.st")
+    tar = os.path.join(up, "%s.blank_program.st" % gen_uid())
+    shutil.copyfile(st, tar)
+    print("Copied: %s" % tar)
+
+    ## archive
+    conn.close()
+
+    ark = os.path.join(workspace, "archive")
+    if not os.path.exists(ark):
+        os.mkdir(ark)
+    fn = "%s.%s" % (gen_uid(), os.path.basename(database))
+    dbak = os.path.join(ark, fn)
+    shutil.copyfile(database, dbak)
+    print("Copied: %s" % dbak)
+
+
+
+    print("DONE :-)")
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Export sqlite to workspace folder")
+
+    parser.add_argument("-w", "--workspace",
+                        help="Path to work directory",
+                        action="store", type=str)
+
+    parser.add_argument("-d", "--database",
+                        help="Path to sqlite db",
+                        action="store", type=str)
+
+    parser.add_argument("-o", "--openplc",
+                        help="Path to openplc checkout",
+                        action="store", type=str)
+
+    args = parser.parse_args()
+
+    if args.database == None:
+        print("ERROR: need a -d database")
+        parser.print_help()
+        sys.exit(0)
```

Comment:
This is an error condition, so this should be exit(-1). However, an even better way to solve this would be to define it with the argument.

def sql_db(arg_value):
    if not os.path.isfile(string):
        raise ValueError("Something nice here")
    return string

parser.add_argument("-d", "--database", required=True, type=sql_db)

You can do the same for the directory ones, but share the implementation. Then you know you have not just an argument that exists, but one that is valid. That would save the checks you have in your other function.